modules/web-console/frontend/app/configuration/generator/generator/JavaTransformer.service.js

```diff
@@ -798,125 +799,8 @@ export default class IgniteJavaTransformer extends AbstractTransformer {
-        });
-
-        imports.push(...this.collectPropertiesImports(bean.properties));
-
-        if (_.includes(STORE_FACTORY, bean.clsName))
-            imports.push('javax.sql.DataSource', 'javax.cache.configuration.Factory');
-
-        return imports;
-    }
-
-    /**
-     * @param {Array.<Object>} props
-     * @returns {Array.<String>}
-     */
-    static collectPropertiesImports(props) {
-        const imports = [];
-
-        _.forEach(props, (prop) => {
-            switch (prop.clsName) {
-                case 'DATA_SOURCE':
-                    imports.push(prop.value.clsName);
-
-                    break;
-                case 'PROPERTY':
-                case 'PROPERTY_CHAR':
-                case 'PROPERTY_INT':
-                    imports.push('java.io.InputStream', 'java.util.Properties');
-
-                    break;
-                case 'BEAN':
-                    imports.push(...this.collectBeanImports(prop.value));
-
-                    break;
-                case 'ARRAY':
-                    if (!prop.varArg)
-                        imports.push(prop.typeClsName);
-
-                    if (this._isBean(prop.typeClsName))
-                        _.forEach(prop.items, (item) => imports.push(...this.collectBeanImports(item)));
-
-                    if (prop.typeClsName === 'java.lang.Class')
-                        _.forEach(prop.items, (item) => imports.push(item));
-
-                    break;
-                case 'COLLECTION':
-                    imports.push(prop.typeClsName);
-
-                    if (this._isBean(prop.typeClsName)) {
-                        _.forEach(prop.items, (item) => imports.push(...this.collectBeanImports(item)));
-
-                        imports.push(prop.implClsName);
-                    }
-                    else if (prop.implClsName === 'java.util.ArrayList')
-                        imports.push('java.util.Arrays');
-                    else
-                        imports.push(prop.implClsName);
-
-                    break;
-                case 'ENUM_COLLECTION':
-                    imports.push(prop.typeClsName);
-
-                    if (prop.implClsName === 'java.util.ArrayList')
-                        imports.push('java.util.Arrays');
-                    else
-                        imports.push(prop.implClsName);
-
-                    break;
-                case 'MAP':
-                    imports.push(...this._collectMapImports(prop));
-
-                    break;
-                default:
-                    if (!this.javaTypesNonEnum.nonEnum(prop.clsName))
-                        imports.push(prop.clsName);
-            }
-        });
-
-        return imports;
-    }
-
     static _prepareImports(imports) {
-        return _.sortedUniq(_.sortBy(_.filter(imports, (cls) => !_.startsWith(cls, 'java.lang.') && _.includes(cls, '.'))));
+        return _.sortBy(_.filter(imports, (cls) => !_.startsWith(cls, 'java.lang.') && _.includes(cls, '.')));
```

Comment:
Am I correct to assume that this function is always called after `collectConfigurationImports`? If true, I think it would make sense to move it into `collectConfigurationImports`.

Reply:
It is used for imports and static imports.